pkgs/top-level/splice.nix

```diff
@@ -0,0 +1,76 @@
+# The `splicedPackages' package set, and its use by `callPackage`
+#
+# The `buildPackages` pkg set is a new concept, and the vast majority package
+# expression (the other *.nix files) are not designed with it in mind. This
+# presents us with a problem with how to get the right version (build-time vs
+# run-time) of a package to a consumer that isn't used to thinking so cleverly.
```

Comment:
True, but by that token would could just have every package just take `{pkgs, buildPackages}: ..`. I...don't actually see a problem with that, but would go against the `callPackage` style.

Separately, since most of the time the package has a build- or run- intention, people have talked about making a "curated" splicing with sane defaults.